File: java-backend/src/main/java/com/seekho/api/mapper/AnnouncementMapper.java

package com.seekho.api.mapper;

import com.seekho.api.dto.AnnouncementDto;
import com.seekho.api.entity.Announcement;

public class AnnouncementMapper {

    public static AnnouncementDto toDto(Announcement entity) {
        AnnouncementDto dto = new AnnouncementDto();
        dto.setaId(entity.getaId());
        dto.setaDesc(entity.getaDesc());
        dto.setaCreatedAt(entity.getaCreatedAt());
        dto.setaIsActive(entity.getaIsActive());
        return dto;
    }

    public static Announcement toEntity(AnnouncementDto dto) {
        Announcement entity = new Announcement();
        entity.setaId(dto.getaId());  // May be null for new records
        entity.setaDesc(dto.getaDesc());
        entity.setaCreatedAt(dto.getaCreatedAt());  // Optional
        entity.setaIsActive(dto.getaIsActive());
        return entity;
    }
}

File: dotnet-backend/Mappers/AnnouncementMapper.cs
using dotnet_backend.Models;
using dotnet_backend.DTOs;

namespace dotnet_backend.Mappers
{
    public static class AnnouncementMapper
    {
        public static AnnouncementDTO ToDTO(Announcement announcement)
        {
            return new AnnouncementDTO
            {
                AnnouncementId = announcement.AnnouncementId,
                AnnouncementText = announcement.AnnouncementText
            };
        }

        public static Announcement ToEntity(AnnouncementDTO dto)
        {
            return new Announcement
            {
                AnnouncementId = dto.AnnouncementId,
                AnnouncementText = dto.AnnouncementText
            };
        }
    }
}

